### DISCUSSION TITLE
@umijs/plugin-qiankun支持主应用为antd pro项目(umi项目)，子应用非umi项目吗（比如webpack 或者vue项目）？

### DISCUSSION THREAD

**Discussion by kgloveyou** [umijs/umi#12102] [Q&A]
@umijs/plugin-qiankun支持主应用为antd pro项目(umi项目)，子应用非umi项目吗（比如webpack 或者vue项目）？

**Comment by fz6m** [comment]
1. 现在没有 `@umijs/plugin-qiankun` 这个东西了，这个是 umi 3 的插件，仓库在 [`umijs/plugins`](https://github.com/umijs/plugins) ，umi 3 已经不维护了，建议升级到 umi 4 。

2. umi 4 中的 qiankun 插件在 `@umijs/plugins` 里（ 没有 `@umijs/plugin-qiankun` 了 ），如果你创建一个 umi 4 的 max 项目，qiankun 功能是内置的，即开即用比较方便。

3. qiankun 里有一个 `loadMicroApp` 方法可以去加载非 umi 的子应用，但需要你自己根据 qiankun 文档处理好子应用的配置，publicPath ，路由等，umi 的 qiankun 插件只是把这些都内置起来了不需要你麻烦去处理而已，如果你自己去加载非 umi + qiankun 的应用，就相当于原生的在用 qiankun ，所以需要你自己根据 qiankun 文档探索 qiankun 的使用方式。

**Reply by kgloveyou** [reply]
1、因为十几个应用都是历史项目，基于    `"umi": "^3.5.13"`的，一起去升级风险比较大，时间上也不允许。所以只能考虑`@umijs/plugin-qiankun`插件。
3、回到标题中的问题：主应用是（新建的）antd pro项目(umi项目)，但是子应用（历史项目）有`umi/webpack/vue`多个技术栈。
要使用qiankun微前端的话，可以主应用使用`@umijs/plugin-qiankun`写法，(umi)子应用使用`@umijs/plugin-qiankun`,(webpack/vue)子应用使用原生的`qiankun`写法吗？
还是说，目前这种情况，主应用和子应用全部按照原生的`qiankun`写法呢？

**Reply by fz6m** [reply]
只有 umi 的项目可以用 `@umijs/plugin-qiankun` ( umi 3 ) ，不过你可以在 umi 主应用里用 qiankun 的方法 `loadMicroApp` 去加载非 umi 的子应用，这个子应用必须要遵循 qiankun 的文档去修改好他的配置、导出，就是确保这个子应用是合理的 qiankun 子应用就可以。

**Comment by kgloveyou** [comment]
> 只有 umi 的项目可以用 `@umijs/plugin-qiankun` ( umi 3 ) ，不过你可以在 umi 主应用里用 qiankun 的方法 `loadMicroApp` 去加载非 umi 的子应用，这个子应用必须要遵循 qiankun 的文档去修改好他的配置、导出，就是确保这个子应用是合理的 qiankun 子应用就可以。

`loadMicroApp` 这段代码应该写在哪里呢

**Reply by kgloveyou** [reply]
loadMicroApp 这段代码应该写在哪里呢

**Reply by fz6m** [reply]
1. umi 3 不维护了，建议升级到 umi 4 。

2. umi 4 使用 `loadMicroApp` 加载子应用的示例在 [examples/qiankun-master](https://github.com/umijs/umi/blob/0c561eab4f623cf38366889c8fdb2468fe7a17fa/examples/qiankun-master/pages/manual-slave.tsx#L9) 。

3. 关于 `loadMicroApp` 在 umi 3 的用法，请自行参考 qiankun 文档学习和研究。
